internal/server/event_list.go

```diff
@@ -0,0 +1,70 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package server
+
+import (
+	"context"
+	"fmt"
+
+	irievent "github.com/ironcore-dev/ironcore/iri/apis/event/v1alpha1"
+	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func (s *Server) filterEvents(events []*irievent.Event, filter *iri.EventFilter) []*irievent.Event {
+	if filter == nil {
+		return events
+	}
+
+	var (
+		res []*irievent.Event
+		sel = labels.SelectorFromSet(filter.LabelSelector)
+	)
+	for _, iriEvent := range events {
+		if !sel.Matches(labels.Set(iriEvent.Spec.InvolvedObjectMeta.Labels)) {
+			continue
+		}
+
+		if filter.EventsFromTime > 0 && filter.EventsToTime > 0 {
+			if iriEvent.Spec.EventTime < filter.EventsFromTime || iriEvent.Spec.EventTime > filter.EventsToTime {
+				continue
+			}
+		}
+
+		res = append(res, iriEvent)
+	}
+	return res
+}
+
+func (s *Server) listEvents() ([]*irievent.Event, error) {
+	events := s.eventStore.ListEvents()
+
+	var iriEvents []*irievent.Event
+	for _, event := range events {
```

Comment:
i think this require comment because ListEvents return []*irievent.Event type too

Reply:
i don't see big benefits in this code. You break reference in evenStore.ListEvents(). Why do you have this for loop here?

Reply:
whole this function doesn't make sense. You can cool eventStore.ListEvents directly on line 60

Reply:
I agree with you. I had broken the reference in eventStore.ListEvents and forgot to remove it from here. I will take it out. Thanks for pointing that out 🙂.

Reply:
removed